02-express-tutorial/app.js

```diff
@@ -1 +1,46 @@
-console.log('Express Tutorial')
+const express = require("express");
+const app = express();
+const { products } = require("./data");
+
+app.listen(3000, () => {
+  console.log("server is listening on port 3000");
+});
+
+app.get("/api/v1/products", (req, res) => {
+  res.json(products);
+});
+
+app.get("/api/v1/products/:productID", (req, res) => {
+  const idToFind = parseInt(req.params.productID);
+  const product = products.find((p) => p.id === idToFind);
+  if (!product) {
+    res.status(404).json({ message: "That product was not found." });
+  } else {
+    res.json(product);
+  }
+});
+
+app.get("/api/v1/query", (req, res) => {
+  const { search, limit } = req.query;
```

Comment:
You have to define the maxPrice here. I'm receiving the error "maxPrice is not defined" when accessing the URL http://localhost:3000/api/v1/query?search=al&limit=1.